dapps/assets/src/components/contacts/EditContact.ts

```diff
@@ -0,0 +1,65 @@
+/*
+  Copyright (C) 2018-present evan GmbH.
+
+  This program is free software: you can redistribute it and/or modify it
+  under the terms of the GNU Affero General Public License, version 3,
+  as published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see http://www.gnu.org/licenses/ or
+  write to the Free Software Foundation, Inc., 51 Franklin Street,
+  Fifth Floor, Boston, MA, 02110-1301 USA, or download the license from
+  the following URL: https://evan.network/license/
+*/
+
+// vue imports
+import Component, { mixins } from 'vue-class-component';
+
+// evan.network imports
+import { EvanComponent, SwipePanelComponentClass } from '@evan.network/ui-vue-core';
+import { Prop, Watch } from 'vue-property-decorator';
+import { Contact } from './ContactInterfaces';
+
+@Component
+export default class EditContactComponent extends mixins(EvanComponent) {
+  @Prop() contact: Contact;
+
+  canSubmit = false;
+
+  note: string;
+
+  @Watch('contact', { deep: true })
+  onContactChange(): void {
+    this.note = this.contact.alias;
+  }
+
+  onSubmit(): void {
+    this.closePanel();
+    const updatedContact = { ...this.contact, alias: this.note };
+    this.$emit('update-contact', updatedContact);
+  }
+
+  onNoteChange(): void {
+    this.canSubmit = true;
```

Comment:
tis would not cover if I change the alias and set it later to the same string. Shall we handle this?

Reply:
maybe make a getter for canSubmit() {return this.note !== this.contact.alias;}

Reply:
I fiddled around with a "isDirty" prop, but it became rather complicated. We should add the "dirty" functionality to the evan-form for sure in the future (or look into form libraries). I decided after some trial and error that this should be enough for MVP

Reply:
Dirty flags are currently only handled with the `EvanForm` or `EvanFormControl` stuff. We could adjust the evan-form-control to also handle this flag or passing `EvanFormControl` instances to the `evan-form-control-input`.